app/api/v1/endpoints/datasets.py

```
@@ -41,69 +60,75 @@ async def upload_dataset(data: DatasetUploadRequest):
-            dataset_id=dataset_id,
-            name=dataset_name
-        )
+        # Только пользователи → сохраняются в PostgreSQL
+        if user_ctx.type == "user":
+            await db.execute(
+                "INSERT INTO datasets (id, user_id, city, transport_type, name) VALUES ($1, $2, $3, $4, $5)",
+                dataset_id, user_ctx.user_id, data.city, data.transport_type, dataset_name
+            )
+        return DatasetUploadResponse(dataset_id=dataset_id)
 
     except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)